Editor/EditorCore/EditorSceneViewPicker.cs

```diff
@@ -83,7 +83,17 @@ public static ProBuilderMesh DoMouseClick(Event evt, SelectMode selectionMode, S
                     UndoUtility.RecordSelection(mesh, "Select Face");
 
                     if (sel > -1)
-                        mesh.RemoveFromFaceSelectionAtIndex(sel);
+                    {
+                        if (!appendModifier || s_Selection.face == mesh.GetActiveFace())
+                        {
+                            mesh.RemoveFromFaceSelectionAtIndex(sel);
+                        }
+                        else
+                        {
+                            mesh.selectedFaceIndicesInternal = mesh.selectedFaceIndicesInternal.Remove(ind);
+                            mesh.SetSelectedFaces(mesh.selectedFaceIndicesInternal.Add(ind));
+                        }
+                    }
```

Comment:
If `SelectionModifierBehaviour` now changes the behavior of mouse clicking, we should support all cases. ex,

```
                    if (sel > -1)
                    {
                        if (pickerPreferences.selectionModifierBehavior == SelectionModifierBehavior.Add)
                        {
                            // Make the selected face 'active' by making it the last in the selection array
                            mesh.RemoveFromFaceSelectionAtIndex(sel);
                            mesh.AddToFaceSelection(ind);
                        }
                        else
                        {
                            // If the face is in selection, appendModifier is true. In this case Subtract and Difference
                            // provide the same result.
                            mesh.RemoveFromFaceSelectionAtIndex(sel);
                        }
                    }
                    else
                    {
                        mesh.AddToFaceSelection(ind);
                    }```

Reply:
Note - it looks like we aren't treating mouse click vertex selection as shared vertices. We  should be.

Reply:
The selection modifier behavior is for drag selection and is documented as such. Do we want to change the picking behavior in the same way? As per @gabrielw-us he would like the behavior of the ctrl+click to be substract and shift to be add. Doesn't this conflict with using the preference for the shift?